mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp

```diff
@@ -128,12 +133,50 @@ extern "C" MiirHandle miirCreateHandle(const char *arguments) {
 
   ModuleOp module = handle->getModule();
 
-  if (failed(conv2dGenerator.genConvModule(module, builder))) {
+  if (failed(conv2dGenerator.genConvModule(module))) {
     return nullptr;
   }
   return handle;
 }
 
+extern "C" MiirHandle miirCreateHandleWithModule(MlirModule module,
+                                                 const char *arch) {
+  const std::lock_guard<std::mutex> lock(mutex);
+
+  std::string triple;
+  std::string chip;
+  std::string features;
+  IsaNameParser parser(arch);
+  if (failed(parser.parseIsaName(chip, triple, features))) {
+    return nullptr;
+  }
+
+  ModuleOp m = unwrap(module);
+  MiirHandle_s *handle = new MiirHandle_s(m);
+
+  // Conv2dGenerator conv2dGenerator; //(chip, triple, features);
+
+  // walk module and test conv2d
+  // const auto &config = conv2dGenerator.getConfig();
+  // if (failed(MIOpenEnabled(config))) {
+  //   return nullptr;
+  // }
+
+  handle->triple = triple;
+  handle->chip = chip;
+  handle->features = features;
+  // handle->perfConfig = config.perfConfig;
+
+  // count kernels from module funcs
+  handle->kernelCount = 0;
+  m.walk([&](FuncOp func) {
+    if (func->hasAttr("kernel"))
+      handle->kernelCount++;
+  });
```

Comment:
But since we already use that API to get kernel count, why invent a new way to do it? When the assumption is not right, it is confusing to return a wrong value. e.g, at that scenario, `miirGetKernelCount` will return this "possibly wrong" kernel count from the handle.